ui/mainwindow.cpp

```diff
@@ -764,11 +764,13 @@ void MainWindow::setupPathSelectionComboBox() {
     ui->pathSelectionComboBox2->clear();
 
     QStringList comboBoxItems;
-    for (QMap<QString, Path*>::key_iterator it = g_assemblyGraph->m_deBruijnGraphPaths.keyBegin();
-         it != g_assemblyGraph->m_deBruijnGraphPaths.keyEnd(); ++it)
-    {
-        comboBoxItems.push_back(*it);
+    std::string key_buffer;
+    for (auto it = g_assemblyGraph->m_deBruijnGraphPaths.begin();
```

Comment:
is it possible to write something like that?
```cpp
for (const auto &[key, _] : g_assemblyGraph->m_deBruijnGraphPaths) {
    comboBoxItems.push_back(key);
}
```

Reply:
sadly, no. Iterators here are not pairs of keys / values, as it's quite special string-key map

Reply:
Ok. According to the documentation, there is `std::string key() const` method of iterator. I think it will be much simpler to use `it.key()` without `key_buffer`.

Reply:
You're quite right. However in this case we will create new string every time. The way above allows to save on memory allocation / deallocation.

Reply:
Maybe now with `parallel_flat_hash_map` it's possible? :)

Reply:
No. The map will change in the next PR :)